qiskit/providers/backend_compat.py

```diff
@@ -13,24 +13,367 @@
 from qiskit.providers.models.backendproperties import BackendProperties
+
 from qiskit.providers.models.pulsedefaults import PulseDefaults
 from qiskit.providers.options import Options
 from qiskit.providers.exceptions import BackendPropertyError
+from qiskit.providers.models.backendproperties import Gate as GateSchema
+
+
+logger = logging.getLogger(__name__)
+
+
+class IBMQubitProperties(QubitProperties):
+    """A representation of the properties of a qubit on an IBM backend."""
+
+    __slots__ = (  # pylint: disable=redefined-slots-in-subclass
+        "t1",
+        "t2",
+        "frequency",
+        "anharmonicity",
+        "operational",
+    )
+
+    def __init__(  # type: ignore[no-untyped-def]
+        self,
+        t1=None,
+        t2=None,
+        frequency=None,
+        anharmonicity=None,
+        operational=True,
+    ):
+        """Create a new ``IBMQubitProperties`` object
+
+        Args:
+            t1: The T1 time for a qubit in secs
+            t2: The T2 time for a qubit in secs
+            frequency: The frequency of a qubit in Hz
+            anharmonicity: The anharmonicity of a qubit in Hz
+            operational: A boolean value representing if this qubit is operational.
+        """
+        super().__init__(t1=t1, t2=t2, frequency=frequency)
+        self.anharmonicity = anharmonicity
+        self.operational = operational
+
+    def __repr__(self):  # type: ignore[no-untyped-def]
+        return (
+            f"IBMQubitProperties(t1={self.t1}, t2={self.t2}, frequency={self.frequency}, "
+            f"anharmonicity={self.anharmonicity})"
+        )
 
 
 def convert_to_target(
+    configuration: Union[QasmBackendConfiguration, PulseBackendConfiguration],
+    pulse_defaults: Optional[Dict] = None,
+    properties: Optional[Dict] = None,
+):
+    """Decode transpiler target from backend data set.
+
+    This function directly generates ``Target`` instance without generating
+    intermediate legacy objects such as ``BackendProperties`` and ``PulseDefaults``.
+
+    Args:
+        configuration: Backend configuration.
+        pulse_defaults: Backend pulse defaults dictionary.
+        properties: Backend property dictionary.
+
+    Returns:
+        A ``Target`` instance.
+    """
+
+    # importing pacakges where they are needed, to avoid cyclic-import.
+    from qiskit.transpiler.target import Target
+    from qiskit.circuit.controlflow import ForLoopOp, IfElseOp, SwitchCaseOp, WhileLoopOp
+    from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
+    from qiskit.qobj.pulse_qobj import PulseLibraryItem
+    from qiskit.qobj.converters.pulse_instruction import QobjToInstructionConverter
+    from qiskit.providers.models.pulsedefaults import Command
+    from qiskit.pulse.calibration_entries import PulseQobjDef
+    from qiskit.circuit.parameter import Parameter
+    from qiskit.circuit.gate import Gate
```

Comment:
All of them leads to cyclic-import?

Reply:
Only `from qiskit.transpiler.target import Target` causes the cyclic-import problem.
But, since these imports are only used in `convert_to_target` I have kept it into this function only.
This is because, had I put it outside(on the top) it would cause the same cyclic-import in the future if any other function finds a way into this file which could use imports that cause cyclic-import problems, so the best solution here is to do imports at the site of usage! :)
I would change these if you say so :)